otelconf/resource_test.go

```diff
@@ -0,0 +1,89 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package otelconf
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/sdk/resource"
+	semconv "go.opentelemetry.io/otel/semconv/v1.37.0"
+)
+
+func TestNewResource(t *testing.T) {
+	tests := []struct {
+		name         string
+		config       OpenTelemetryConfigurationResource
+		wantResource *resource.Resource
+		wantErrT     error
+	}{
+		{
+			name:         "no-resource-configuration",
+			wantResource: resource.Default(),
+		},
+		{
+			name:         "invalid resource",
+			config:       "",
+			wantResource: nil,
+			wantErrT:     newErrInvalid("resource"),
+		},
+		{
+			name:         "resource-no-attributes",
+			config:       &ResourceJson{},
+			wantResource: resource.NewSchemaless(),
+		},
+		{
+			name: "resource-with-schema",
+			config: &ResourceJson{
+				SchemaUrl: ptr(semconv.SchemaURL),
+			},
+			wantResource: resource.NewWithAttributes(semconv.SchemaURL),
+		},
+		{
+			name: "resource-with-attributes",
+			config: &ResourceJson{
+				Attributes: []AttributeNameValue{
+					{Name: "service.name", Value: "service-a"},
+				},
+			},
+			wantResource: resource.NewWithAttributes("",
+				semconv.ServiceName("service-a"),
```

Comment:
These could drift. We don't want to test `semconv` stability here.

```suggestion
					{Name: semconv.ServiceNameKey, Value: "service-a"},
				},
			},
			wantResource: resource.NewWithAttributes("",
				semconv.ServiceName("service-a"),
```

Similar below.

Reply:
updated